libclamav/asn1.c

```diff
@@ -1620,7 +1620,7 @@ static cl_error_t asn1_parse_mscat(struct cl_engine *engine, fmap_t *map, size_t
                         cli_dbgmsg("asn1_parse_mscat: Found Authenticode certificate blocked by %s\n", crt->name ? crt->name : "(unnamed CRB rule)");
                         if (NULL != ctx) {
                             ret = cli_append_virus(ctx, crt->name ? crt->name : "(unnamed CRB rule)");
-                            if ((ret == CL_VIRUS) && !SCAN_ALLMATCHES) {
+                            if (ret == CL_VIRUS) {
                                 crtmgr_free(&newcerts);
```

Comment:
There are a couple places in this function with this particular free.  Do you think it's worth it to move this to the finish block so that it is only done in one place?

Reply:
Sure can do. 

I'm afraid of doing the clean up on the RTF parser, though. 😅